File: Practice/10/C++/fixed_sum.cpp
#include <iostream>

int main() {
    long s, l1, r1, l2, r2;

	std::cout << "Введите сумму, границы первого промежутка и границы второго промежутка на одной строке через пробел" << std::endl;
    std::cin >> s >> l1 >> r1 >> l2 >> r2;

    long long i{l1};
    long long j{r2};

	if (i + j < s) i = min(r1, s-j);
	else           j = max(l2, s-i);

	if (i + j < s) i = min(r1, s-j);
	else           j = max(l2, s-i);

    std::string out{(i + j == s)
        ? "x1 = " + std::to_string(std::min(i, j)) +
		  "\nx2 = " + std::to_string(std::max(i, j))
        : "Нет таких x."};

    std::cout << out << std::endl;
}

File: Practice/10/Python/fixed_sum.py
print("Введите сумму, границы первого промежутка и границы второго промежутка на одной строке через пробел")
s, l1, r1, l2, r2 = [int(x) for x in input().split()]

i = l1
j = r2

if (i + j < s):
    i = min(r1, s-j)
else:
    j = max(l2, s-i)

if (i + j < s):
    i = min(r1, s-j)
else:
    j = max(l2, s-i)

if i + j == s:
    print("x1 =", min(i, j), "\nx2 =", max(i, j))
else:
    print("Нет таких x")
